src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java

```diff
@@ -701,6 +702,29 @@ public synchronized void setApprovedSignatures(String[] signatures) throws IOExc
         return aclApprovedSignatures.toArray(new String[aclApprovedSignatures.size()]);
     }
 
+    @DataBoundSetter
+    public synchronized void setApprovedScriptHashes(String[] scriptHashes) throws IOException {
+        Jenkins.getInstance().checkPermission(Jenkins.RUN_SCRIPTS);
+        approvedScriptHashes.clear();
+        List<String> goodScriptHashes = new ArrayList<>(scriptHashes.length);
+        Pattern sha1Pattern = Pattern.compile("^[a-fA-F0-9]{40}$");
```

Comment:
```suggestion
        Pattern sha1Pattern = Pattern.compile("[a-fA-F0-9]{40}");
```

should suffice since you are using `matches` not `find`.